bgs/compactor.go

```diff
@@ -263,34 +252,37 @@ func (c *Compactor) doWork(bgs *BGS) {
 	}
 }
 
-func (c *Compactor) compactNext(ctx context.Context, bgs *BGS) (*CompactorState, error) {
+func (c *Compactor) compactNext(ctx context.Context, bgs *BGS) (state CompactorState, err error) {
```

Comment:
if i could delete one thing from Go, it would be named returns. I really try to avoid using them ever as they always make the code harder to follow